lib/elixir/lib/exception.ex

```diff
@@ -683,6 +691,10 @@ defmodule UndefinedFunctionError do
     "      * " <> fun <> "/" <> Integer.to_string(arity) <> "\n"
   end
 
+  defp macro_with_same_mfa?(module, function, arity) do
+    function_exported?(module, :__info__, 1) and macro_exported?(module, function, arity)
```

Comment:
@josevalim I'm doing the `function_exported?/3` check here because `macro_exported?/3` will fail when checking Erlang modules:

```
iex> macro_exported?(:erlang, :now, 0)
** (UndefinedFunctionError) function :erlang.__info__/1 is undefined or private
```

Are we fine with this behaviour? Me and @lexmag quickly discussed about this and maybe it would make sense to handle Erlang modules in `macro_exported?/3` and return `false` for them (same as we return `false` for non-loaded modules). If we don't go with this route, we should document that `macro_exported?/3` fails with Erlang modules at least. Wdyt?